ui/src/app/pages/artifacts/components/uploadForm/uploadForm.tsx

```diff
@@ -158,13 +160,46 @@ export class UploadArtifactForm extends PureComponent<UploadArtifactFormProps, U
                         data-testid="form-upload"
                         type="text"
                         filename={this.state.contentFilename}
-                        value={this.state.content}
-                        isRequired={true}
+                        value={this.state.content!}
+                        isRequired={false}
                         allowEditingUploadedText={true}
                         onChange={this.onContentChange}
                         onReadStarted={this.onFileReadStarted}
                         onReadFinished={this.onFileReadFinished}
                         isLoading={this.state.contentIsLoading}
+                        isDisabled={!!this.state.fromURL}
+                    />
+                    <span>or:</span>
+                    <TextInput
+                        className="artifact-url"
+                        isRequired={false}
+                        type="text"
+                        id="form-url"
+                        data-testid="form-url"
+                        name="form-url"
+                        aria-describedby="form-url-helper"
+                        value={this.state.fromURL ? this.state.fromURL : ""}
+                        placeholder="URL of the artifact"
+                        onChange={this.onURLChange}
+                        isDisabled={!!this.state.content}
+                    />
+                </FormGroup>
+                <FormGroup
+                    label="Artifact SHA"
+                    isRequired={false}
+                    fieldId="form-artifact-sha"
+                >
+                    <TextInput
+                        className="artifact-sha"
+                        isRequired={false}
+                        type="text"
+                        id="form-sha"
+                        data-testid="form-sha"
+                        name="form-sha"
+                        aria-describedby="form-sha-helper"
+                        value={this.state.sha ? this.state.sha : ""}
+                        placeholder="SHA256 of the artifact"
+                        onChange={this.onShaChange}
                     />
```

Comment:
This is good for now, but FYI I might change this UI experience to work more like what we have now in API Designer:

![image](https://user-images.githubusercontent.com/1890703/185475586-11828247-dee9-4199-8b66-dab07776eab2.png)

This way the user can validate the content in the browser before submitting it.  And we lessen the opportunities for users to send us URLs that might not respond quickly.

And it makes things more consistent across services.

Reply:
Nothing to change for now on this.  :)

Reply:
Nice 👍